### PR TITLE
MODUSERBL-208: Query with username fails when username contains non-url-encoded characters

### PR DESCRIPTION
**Pull request by kapil-epam**
https://folio-org.atlassian.net/browse/MODUSERBL-208

### CODE REVIEW

**Review by steveellis — APPROVED** [review]
Thanks I think this is sufficient. @kapil-epam note that if we backport this to P and Q we will also need to backport the code from https://github.com/folio-org/mod-users-bl/pull/185 for it all to work.

cc @gurleenkaurbp

File: src/main/java/org/folio/rest/impl/BLUsersAPI.java
[the remark this answers is not in the document]
Comment (kapil-epam) [reply src/main/java/org/folio/rest/impl/BLUsersAPI.java:1]: Hi @julianladisch : Pls check the fix in this commit: https://github.com/folio-org/mod-users-bl/pull/204/commits/32a91e0c595b9cf0543abe6dad619b320f132af1

File: src/test/java/org/folio/rest/BLUsersAPITest.java [hunk 1]
```diff
@@ -331,6 +347,71 @@ public void getBlUsersSelfWithoutUserId(TestContext context) {
...
+  public void getBlUsersSelfWithURLEncodedUsername(TestContext context) {
+    Header header = new Header(RestVerticle.OKAPI_HEADER_TOKEN, getTokenWithoutUserId(
+      URLEncoder.encode("myuserñ`ame", StandardCharsets.UTF_8).toString(), "diku"));
+    given().
+      spec(okapi).port(port).header(header).
+      when().
+      get("/bl-users/_self").
+      then().
+      statusCode(200);
+  }
```
[the remark this answers is not in the document]
Comment (kapil-epam) [reply src/test/java/org/folio/rest/BLUsersAPITest.java]: fixed=> https://github.com/folio-org/mod-users-bl/pull/204/commits/4d51d6f838f000a0e0a2ca57088ccf098b2bbd38

File: src/test/java/org/folio/rest/BLUsersAPITest.java [hunk 2]
```diff
@@ -331,6 +347,71 @@ public void getBlUsersSelfWithoutUserId(TestContext context) {
...
+
+  @Test
+  public void getBlUserByUserNameWhenPassingURLEncodedUsername(TestContext context) {
+    given().
+      spec(okapi).port(port).
+      when().
+      get("/bl-users/by-username/" + URLEncoder.encode("myuserñ`ame", StandardCharsets.UTF_8)).
+      then().
+      statusCode(200);
+  }
```
[the remark this answers is not in the document]
Comment (kapil-epam) [reply src/test/java/org/folio/rest/BLUsersAPITest.java]: fixed=> https://github.com/folio-org/mod-users-bl/pull/204/commits/4d51d6f838f000a0e0a2ca57088ccf098b2bbd38

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Pull request merged**